tango_ros_common/tango_ros_native/include/tango_ros_native/tango_ros_node.h

```diff
@@ -174,10 +174,13 @@ class TangoRosNode {
   std::condition_variable color_image_available_;
 
   double time_offset_ = 0.; // Offset between tango time and ros time in s.
+  int localization_mode_;
+  bool first_localized_ = false;
+  bool localization_lost_ = false;
```

Comment:
AFAIU, you are implementing a small state machine right? 
Perhaps I would just declare a variable that represented the status of the localization. For example, an `enum` with three states: starting, localized, localization lost (or more proper names) and take actions according to that state instead of having multiple variables.